testsuite/tests/singlecluster/authorino/authorization/spicedb/test_spicedb.py

```diff
@@ -44,55 +60,43 @@ def relationship_config():
 
 
 @pytest.fixture(scope="module")
-def spicedb_info(testconfig, skip_or_fail):
+def secret(request, cluster, module_label, kuadrant, blame):
     """
-    Validates that only the 'spicedb' section of the test config is used,
-    and returns the relevant configuration (e.g., URL, secret, etc.).
+    Creates a secret containing the SpiceDB preshared key for Authorino to use.
+    The secret is created in Authorino's namespace.
     """
-    try:
-        testconfig.validators.validate(only="spicedb")
-    except (KeyError, ValidationError) as exc:
-        skip_or_fail(f"Spicedb configuration item is missing: {exc}")
-    return testconfig["spicedb"]
-
-
-@pytest.fixture(scope="module")
-def create_secret(request, cluster, blame, spicedb_info, module_label, testconfig, kuadrant):
-    """Create a secret for Authorino to use in authorization with SpiceDB. Secret mut be created in namespace where
-    Authorino is deployed."""
-    secret_name = blame("spicedb")
     if kuadrant:
-        cluster = cluster.change_project(testconfig["service_protection"]["system_project"])
+        cluster = cluster.change_project(settings["service_protection"]["system_project"])
     secret = Secret.create_instance(
         cluster,
-        secret_name,
-        {"grpc-preshared-key": spicedb_info["password"]},
+        blame("spicedb"),
+        stringData={"grpc-preshared-key": "secret"},
         labels={"app": module_label},
     )
     request.addfinalizer(secret.delete)
     secret.commit()
-    return secret_name
+    return secret
 
 
 @pytest.fixture(scope="module")
-def spicedb(spicedb_info, schema_config):
+def spicedb(spicedb_deployment, schema_config):
     """
-    Creates a SpiceDB service instance used for sending authorization requests.
+    Creates a SpiceDB client instance for managing schemas and relationships.
     After the tests, it clears all relationships.
     """
-    service = SpiceDBService(spicedb_info["url"], spicedb_info["password"])
+    service = SpiceDBClient(spicedb_deployment.grpc_url, spicedb_deployment.preshared_key)
```

Comment:
spicedb client is fully dependent on spicedb deployment. You can make spicedb class to return you his client, instead of building it yourself here